Tests/Marketplace/search_and_install_packs.py

```diff
@@ -417,77 +362,118 @@ def search_pack_and_its_dependencies(client: demisto_client,
             Each list contain one pack and its dependencies.
-
     """
-    pack_data = {}
-    if pack_id not in packs_to_install:
-        pack_display_name = get_pack_display_name(pack_id)
-        if pack_display_name:
-            pack_data = search_pack(client, pack_display_name, pack_id, lock)
-        if pack_data is None:
-            pack_data = {
-                'id': pack_id,
-                'version': '1.0.0'
-            }
-
-    if pack_data:
-        dependencies = get_pack_dependencies(client, pack_data, lock)
-
-        current_packs_to_install = [pack_data]
-        if dependencies:
-            # Check that the dependencies don't include a deprecated pack:
-            for dependency in dependencies:
-                pack_path = os.path.join(PACKS_FOLDER, dependency.get('id'))
-                if is_pack_deprecated(pack_path):
-                    logging.critical(f'Pack {pack_id} depends on pack {dependency.get("id")} which is a deprecated '
-                                     f'pack.')
-                    global SUCCESS_FLAG
-                    SUCCESS_FLAG = False
-                else:
-                    current_packs_to_install.append(dependency)
-
-        lock.acquire()
-        if one_pack_and_its_dependencies_in_batch:
-            pack_and_its_dependencies = \
-                {p['id']: p for p in current_packs_to_install if p['id'] not in packs_in_the_list_to_install}
-            if pack_and_its_dependencies:
-                packs_in_the_list_to_install += pack_and_its_dependencies
-                batch_packs_install_request_body.append(list(pack_and_its_dependencies.values()))  # type:ignore[union-attr]
-        else:
-            for pack in current_packs_to_install:
-                if pack['id'] not in packs_to_install:
-                    packs_to_install.append(pack['id'])
-                    installation_request_body.append(pack)
-        lock.release()
+    if is_post_update:
+        # On post-update, we want to check for deprecation status locally before making the API call,
+        # because if the pack has been deprecated, the test upload flow won't upload the pack to the bucket,
+        # and the API call will fail.
+        logging.info(f"Detected post-update run mode ('BUCKET_UPLOAD'='{is_post_update}'. "
+                     f"Pack deprecation status will be checked using local pack metadata.")
+
+        if is_pack_deprecated(pack_id=pack_id, check_locally=True):
+            logging.warning(f"Pack '{pack_id}' is deprecated (hidden) and will not be installed.")
+            return
+
+    else:
+        # On pre-update, we use current prod data, so deprecated packs are not supposed to be passed to this function.
+        # If they are - the API call will fail with a 400 status "item not found" error.
+        logging.info(f"Detected pre-update run mode ('BUCKET_UPLOAD'='{is_post_update}'. "
```

Comment:
What does `'BUCKET_UPLOAD'='{is_post_update}'` means? Why BUCKET_UPLOAD will be True if it's a post-update in a regular build?

Reply:
This is just a log for debugging, it will look something like:
> Detected pre-update run mode ('BUCKET_UPLOAD'='True'). Packs deprecation status will be checked using Marketplace API.

There is, however, no need to print this variable since there are already different messages if it's true or false, so I'll update that to something like this:
```suggestion
        logging.info(f"Detected pre-update run mode."
```